tree/dataframe/test/dataframe_vary.cxx

```diff
@@ -604,56 +633,227 @@ TEST_P(RDFVary, VariedHistosMustHaveNoDirectory)
    EXPECT_EQ(hs["x:1"].GetDirectory(), nullptr);
 }
 
-TEST_P(RDFVary, DISABLED_VaryDefinePerSample)
+TEST(RDFVary, WithRange) // no Range in multithreaded runs
 {
+   auto h = ROOT::RDataFrame(10)
+               .Define("x", [](ULong64_t e) { return int(e); }, {"rdfentry_"})
+               .Vary(
+                  "x",
+                  [](int x) {
+                     return ROOT::RVecI{x - 5, x + 5};
+                  },
+                  {"x"}, 2)
+               .Range(7)
+               .Filter("x > 1")
+               .Range(3)
+               .Sum<int>("x");
+   auto hs = VariationsFor(h);
 
+   EXPECT_EQ(*h, 9);
+   EXPECT_EQ(hs["nominal"], 9);
+   EXPECT_EQ(hs["x:0"], 0);
+   EXPECT_EQ(hs["x:1"], 18);
 }
 
-TEST_P(RDFVary, DISABLED_WithRange)
+TEST_P(RDFVary, VaryRedefine)
 {
+   // first redefine and then vary
+   auto h = ROOT::RDataFrame(10)
+               .Define("x", [](ULong64_t e) { return int(e); }, {"rdfentry_"})
+               .Redefine("x", [] { return 25; })
+               .Vary(
+                  "x",
+                  [](int x) {
+                     return ROOT::RVecI{x - 1, x + 1};
+                  },
+                  {"x"}, 2)
+               .Sum<int>("x");
+   auto hs = VariationsFor(h);
 
+   EXPECT_EQ(*h, 250);
+   EXPECT_EQ(hs["nominal"], 250);
+   EXPECT_EQ(hs["x:0"], 240);
+   EXPECT_EQ(hs["x:1"], 260);
 }
 
-TEST_P(RDFVary, DISABLED_VaryRedefine)
+TEST_P(RDFVary, RedefineVariedColumn)
 {
-
+   // first vary and then redefine --> not legal
+   EXPECT_THROW(ROOT::RDataFrame(10)
+               .Define("x", [](ULong64_t e) { return int(e); }, {"rdfentry_"})
+               .Vary(
+                  "x",
+                  [](int x) {
+                     return ROOT::RVecI{x - 1, x + 1};
+                  },
+                  {"x"}, 2)
+               .Redefine("x", [] { return 25; })
+               .Sum<int>("x"), std::runtime_error);
```

Comment:
there is too much content in this `EXPECT_THROW`. the test would pass even if we throw a `runtime_error` from the _constructor_. Only the `Redefine` call should be inside the `EXPECT_THROW`